Documentation/gettingstarted/k8s-install-helm.rst

```diff
@@ -0,0 +1,242 @@
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _k8s_install_helm:
+
+***********************
+Installation using Helm
+***********************
+
+This guide will show you how to install Cilium using `Helm
+<https://helm.sh/>`_. This involves a couple of additional steps compared to
+the :ref:`k8s_quick_install` and requires you to manually select the best
+datapath and IPAM mode for your particular environment.
+
+Install Cilium
+==============
+
+.. include:: k8s-install-download-release.rst
+
+.. tabs::
+
+    .. group-tab:: Generic
+
+       These are the generic instructions on how to install Cilium into any
+       Kubernetes cluster using the default configuration options below. Please
+       see the other tabs for distribution/platform specific instructions which
+       also list the ideal default configuration for particular platforms.
+
+       **Default Configuration:**
+
+       =============== =============== ==============
+       Datapath        IPAM            Datastore
+       =============== =============== ==============
+       Encapsulation   Cluster Pool    Kubernetes CRD
+       =============== =============== ==============
+
+       .. include:: requirements-generic.rst
+
+       **Install Cilium:**
+
+       Deploy Cilium release via Helm:
+
+       .. parsed-literal::
+
+          helm install cilium |CHART_RELEASE| \\
+            --namespace kube-system
+
+    .. group-tab:: GCP/GKE
+
+       .. include:: requirements-gke.rst
+
+       **Install Cilium:**
+
+       Extract the Cluster CIDR to enable native-routing:
+
+       .. code-block:: shell-session
+
+          NATIVE_CIDR="$(gcloud container clusters describe "${NAME} --zone "${ZONE}" --format 'value(clusterIpv4Cidr)')"
+          echo $NATIVE_CIDR
+
+       Deploy Cilium release via Helm:
+
+       .. parsed-literal::
+
+          helm install cilium |CHART_RELEASE| \\
+            --namespace kube-system \\
+            --set nodeinit.enabled=true \\
+            --set nodeinit.reconfigureKubelet=true \\
+            --set nodeinit.removeCbrBridge=true \\
+            --set cni.binPath=/home/kubernetes/bin \\
+            --set gke.enabled=true \\
+            --set ipam.mode=kubernetes \\
+            --set nativeRoutingCIDR=$NATIVE_CIDR
+
+       The NodeInit DaemonSet is required to prepare the GKE nodes as nodes are added
+       to the cluster. The NodeInit DaemonSet will perform the following actions:
+
+       * Reconfigure kubelet to run in CNI mode
+       * Mount the eBPF filesystem
+
+    .. group-tab:: Azure/AKS
+
+       .. include:: requirements-aks.rst
+
+       **Create a service principal:**
+
+       In order to allow cilium-operator to interact with the Azure API, a
+       service principal is required. You can reuse an existing service
+       principal if you want but it is recommended to create a dedicated
+       service principal for cilium-operator:
```

Comment:
I would suggest mentioning this should be done per installation, otherwise one might re-use the same SP for several clusters:
```suggestion
       In order to allow cilium-operator to interact with the Azure API, a
       service principal is required. You can reuse an existing service
       principal if you want but it is recommended to create a dedicated
       service principal for each Cilium installation:
```
But then, we should also warn that if one incidentally uses the same `--name` as an existing SP, the previous SP will be overwritten. Note the `Found an existing application instance of "6316910e-005f-41c6-a432-bfcc7e459548". We will patch it` in this example:
```console
➜ az ad sp create-for-rbac --name nbusseneau-test-overwrite
Changing "nbusseneau-test-overwrite" to a valid URI of "http://nbusseneau-test-overwrite", which is the required format used for service principal names
In a future release, this command will NOT create a 'Contributor' role assignment by default. If needed, use the --role argument to explicitly create a role assignment.
Creating 'Contributor' role assignment under scope '/subscriptions/22716d91-fb67-4a07-ac5f-d36ea49d6167'
The output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli
{
  "appId": "6316910e-005f-41c6-a432-bfcc7e459548",
  "displayName": "nbusseneau-test-overwrite",
  "name": "http://nbusseneau-test-overwrite",
  "password": "<a first password>",
  "tenant": "***"
}
➜ az ad sp create-for-rbac --name nbusseneau-test-overwrite
Changing "nbusseneau-test-overwrite" to a valid URI of "http://nbusseneau-test-overwrite", which is the required format used for service principal names
Found an existing application instance of "6316910e-005f-41c6-a432-bfcc7e459548". We will patch it
In a future release, this command will NOT create a 'Contributor' role assignment by default. If needed, use the --role argument to explicitly create a role assignment.
Creating 'Contributor' role assignment under scope '/subscriptions/22716d91-fb67-4a07-ac5f-d36ea49d6167'
  Role assignment already exists.

The output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli
{
  "appId": "6316910e-005f-41c6-a432-bfcc7e459548",
  "displayName": "nbusseneau-test-overwrite",
  "name": "http://nbusseneau-test-overwrite",
  "password": "<a new password, replacing the previous one>",
  "tenant": "***"
}
```

To avoid this happening, I would suggest we switch to either:

- Using `--name cilium-operator-$RANDOM`
- Not providing `--name` ([CLI will generate one if not provided](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az_ad_sp_create_for_rbac-optional-parameters))